doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -462,6 +462,17 @@ Displaying information about notations
 
    :flag:`Printing All` to disable other elements in addition to notations.
 
+.. cmd:: Print Notation @string
+
+   Displays the level and associativity of a notation given by a parsing rule
+   :token:`string`.
```

Comment:
```suggestion
   Displays the level and associativity of a notation given by its previously defined
   notation :token:`string`.
```

The command shows more than just "level and associativity".  Maybe just say "Displays information about ..."?

Reply:
The notation string is used to generate productions in the grammar (sometimes more than one, e.g. for recursive notations) but it is somewhat different from the grammar/"parsing rules", so it seemed a bit imprecise.  IMHO.

How about just "Displays information about previously defined notations."?  There's an example right below this line.  It seems to me that simply _showing_ the output is more readable than _describing_ it in prose.  If we just say "information" then it allows for the possibility that you would see additional info for some notation strings.  Or if we later add more info in some cases, we wouldn't need to change the wording.  It's a useful vagueness.  Easy for the user to discover what it does.

FWIW, the form shown by `Print Grammar constr.` is closest to the actual grammar, though I think that's not so easy for users.

@Alizter Have you tried this for recursive notations?  See [Notations with recursive patterns](https://coq.inria.fr/distrib/current/refman/user-extensions/syntax-extensions.html#notations-with-recursive-patterns) and [Notations with recursive patterns involving binders](https://coq.inria.fr/distrib/current/refman/user-extensions/syntax-extensions.html#notations-with-recursive-patterns-involving-binders).  If the results are interestingly different from the current example, maybe include a recursive notation in the example.

Perhaps it would be helpful to also display the module name and line number of the Notation definition.

Reply:
@jfehrle We are not querying the Notation *definition* but rather the *reservation*. This means that this command is blind to if this notation actually means anything. For that we have `Locate` which is able to locate notations and list their definitions and modules.

Recursive notations do work, see the PrintNotation.v file in the test-suite. The output of print notation even shows what exactly it is working with. If you write `Print Notation "x + y".` it will become `Print Notation "_ + _"` as we do some preprocessing. We would like users however to use the latter version since that is more clear, especially in the presence of symbols which have alphanumeric characters.

Reply:
If it's an important distinction, then shouldn't the documentation be clear that it looks at the reservation?  And briefly mention `Locate`?

Reply:
Being vague has its advantages. However, explicitly describing in words what the command does allows for the possibility that a user discovers the command through keyword searching. Please see my recently pushed draft. I'm unsure it is really optimized for discoverability through keyword searching, though.

Reply:
Well, "level" appears in the documentation 272 times and "associativity" 59 times.  In this case, searching the text is not so helpful.  On the other hand, it would make perfect sense to search for "Notation" in the commands index.